.github/workflows/compilation-checks.yml

```diff
@@ -0,0 +1,31 @@
+# Copyright 2020, Data61, CSIRO (ABN 41 687 119 230)
+#
+# SPDX-License-Identifier: BSD-2-Clause
+
+# Compilation actions to run on pull requests
+
+name: Compilation checks
```

Comment:
Can we find a shorter name? Maybe just "Compile". The shorter the name, the easier it is to see what's happening in the "Checks" section of a PR, esp with matrix builds that tend to generate long combination names.

Reply:
Yep, `Compile` sounds good.

Looking to the future though, the number of checks for `sel4test` would be >>50. So perhaps we should isolate them, meaning this one would be `Kernel compile`, and the future seL4test one would be `seL4test compile`.

Reply:
We could also call the workflow `Kernel`, and the job `compile`. Then it should come out as something like `Kernel / compile (ARM, llvm, py3)`.

Reply:
But happy for you to decide either way.